src/lcm/function_representation.py

```diff
@@ -172,11 +181,11 @@ def _get_lookup_function(
     """
     arg_names = [*axis_names, array_name]
 
-    @with_signature(args=arg_names)
-    def lookup_wrapper(*args: Scalar, **kwargs: Scalar) -> Scalar:
+    @with_signature(args=dict.fromkeys(arg_names, "Array"), return_annotation="Array")
+    def lookup_wrapper(*args: Array, **kwargs: Array) -> Array:
         kwargs = all_as_kwargs(args, kwargs, arg_names=arg_names)
         positions = tuple(kwargs[var] for var in axis_names)
-        arr = cast("Array", kwargs[array_name])
+        arr = kwargs[array_name]
         return arr[positions]
```

Comment:
```suggestion
        return kwargs[array_name][positions]
```

Now that there is no `cast()` any more, this is easier to read.